misc-scripts/xref_mapping/XrefParser/JGI_Parser.pm

```diff
@@ -76,89 +74,60 @@ sub run {
 
     my $xref;
 
-    my ($header, $sequence) = $_ =~ /^>?(.+?)\n([^>]*)/s or warn("Can't parse FASTA entry: $_\n"); 
+    my ($header, $sequence) = $record =~ /^>?(.+?)\n([^>]*)/s or croak("Can't parse FASTA entry: $record\n");
+    my $accession = $header;
 
-      my @attr = split/\|/, $header ; 
-      my $acession = $header ;
-    
     # split header in different ways according to source name : 
-    my ( $version,$label  ) ;      
-    if ($source_name=~m/cint_jgi_v1/) { 
+
+    if ($source_name=~m/cint_jgi_v1/) {
       # header format is  >ci0100146277
-      # we want  146277
-      ($acession = $header)  =~s/\w{6}//;  
-      $version = "JGI 1.0" ;  
-    
-    } elsif ($source_name=~m/cint_aniseed_.*v1/) { 
+      # JGI 1.0
+      # we want accession 146277 from above
+      ($accession = $header) =~s/\w{6}//;
+
+    } elsif ($source_name=~m/cint_aniseed_.*v1/) {
       # header format is  >ci0100146277, we want this 
-      $version = "JGI 1.0" ;  
-
-
-    } elsif ($source_name=~m/cint_jgi_v2/) { 
-      $acession = $attr[2]  ; 
-      $version = "JGI 2.0" ;  
-      $label = $attr[3] ; 
-    } elsif ($source_name=~m/cint_aniseed_.*v2/) { 
-      my $aniseed_prefix = "ci0200" ; 
-      $acession = $aniseed_prefix . $attr[2]  ; 
-      $version = "JGI 2.0" ;  
-
-    }else { 
-      print STDERR "WARNING : The source-name specified ($source_name) in the populate_metatable.sql file is\n" .
-        "WARNING : not matching the differnt cases specified in JGI_Parser.pm - plese\n".
-          "WARNING : edit the parser \n" ; 
-      return 1;
-    } 
-    #print "ACCESSION $acession\n" ;  
+      # JGI 1.0
+
+    } else {
+      croak "WARNING : The source-name specified ($source_name) in the populate_metatable.sql file is\n" .
+            "WARNING : not matching the different cases specified in JGI_Parser.pm - please\n".
+            "WARNING : edit the parser \n" ;
+    }
 
     # make sequence into one long string
     $sequence =~ s/\n//g;
 
     # build the xref object and store it
-    $xref->{ACCESSION}     = $acession;
-    $xref->{LABEL}         = $acession;
+    $xref->{ACCESSION}     = $accession;
+    $xref->{LABEL}         = $accession;
     $xref->{SEQUENCE}      = $sequence;
     $xref->{SOURCE_ID}     = $source_id;
     $xref->{SPECIES_ID}    = $species_id;
     $xref->{SEQUENCE_TYPE} = $self->get_sequence_type();
-    $xref->{STATUS}        = 'experimental';
-
-    # pull cg_name from peptide files as well and create dependent xrefs
-#    if ($self->get_sequence_type() =~ /peptide/) {
-#      my ($cg_name) = $cg =~ /cg_name=(.*)/;
-#      my %dep;
-#      $dep{SOURCE_NAME} = 'JGI__Gene';
-#      $dep{LINKAGE_SOURCE_ID} = $xref->{SOURCE_ID};
-#      $dep{SOURCE_ID} = $celera_gene_source_id;
-#      $dep{ACCESSION} = $cg_name;
-#      push @{$xref->{DEPENDENT_XREFS}}, \%dep; # array of hashrefs
-#    }
-#
+
     push @xrefs, $xref;
 
   }
 
   $file_io->close();
-
+  IO::Handle->input_record_separator( "\n");
```

Comment:
This is probably not needed, but it's good anyway to explicitly reset to default.

Reply:
I'm slightly nervous about a worker running another parser after this one and remembering the IO::Handle package state.